addons/payment_adyen/models/payment.py

```diff
@@ -193,7 +192,7 @@ def _adyen_form_get_tx_from_data(self, data):
             shasign_check = tx.acquirer_id._adyen_generate_merchant_sig_sha256('out', data)
         else:
             shasign_check = tx.acquirer_id._adyen_generate_merchant_sig('out', data)
-        if to_native(shasign_check) != to_native(data.get('merchantSig')):
```

Comment:
There's probably some encoding shenanigan crap which required converting to a native str, I'm guessing one of these is (sometimes?) a bytes and the other not.

Reply:
I guess these can be replaced by a call to `to_text` right?

Reply:
Probably, or just fixing up the encoding to get the same thing on both sides? You may want to check if there's any detail in the original commit (where they were added).

Reply:
In fact, wouldn't it be safer to replace all calls to `to_native` by calls to `to_text`? From reading the code `to_native` just seems like a configurable `to_text`, but by default it seems to perform the same task as `to_text` (which can be kept)

Reply:
Yup. In a P3-only world, to_native and to_text are the same thing.